src/components/authentication-error-dialog/index.tsx

```diff
@@ -0,0 +1,97 @@
+import { useCallback, useEffect, useRef } from 'react'
+import { IconSignIn16 } from '@hashicorp/flight-icons/svg-react/sign-in-16'
+import { IconX16 } from '@hashicorp/flight-icons/svg-react/x-16'
+import { AuthErrors } from 'types/auth'
+import { safeAnalyticsTrack } from 'lib/analytics'
+import useAuthentication from 'hooks/use-authentication'
+import Button from 'components/button'
+import Dialog from 'components/dialog'
+import Text from 'components/text'
+import { guessMethodOfDismissal } from './helpers'
+import s from './authentication-error-dialog.module.css'
+
+/**
+ * A dialog that renders for `AuthErrors.RefreshAccessTokenError` errors.
+ * Provides an option for users to sign in again. When dismissed, automatically
+ * signs the user out.
+ */
+const AuthenticationErrorDialog = () => {
+	const closeButtonRef = useRef<HTMLButtonElement>()
+	const { error, signIn, signOut } = useAuthentication()
+	const isRefreshAccessTokenError = error === AuthErrors.RefreshAccessTokenError
```

Comment:
🤔 should we store this in state to make sure the analytics tracking isn't fired off if this is redefined on every re-render?

That reminds me we could check the segment debugger to see what is firing.